src/dbus/common/dbus_message_helper.hpp

```diff
@@ -313,29 +313,29 @@ template <> struct DBusTypeTrait<BorderRoutingCounters>
 
 template <> struct DBusTypeTrait<Nat64ComponentState>
 {
-    // struct of { uint8, uint8 }
+    // struct of { string, string }
     static constexpr const char *TYPE_AS_STRING = "(ss)";
 };
 
 template <> struct DBusTypeTrait<Nat64TrafficCounters>
 {
-    // struct of { uint64, uint64, uint64, uint64 }
-    static constexpr const char *TYPE_AS_STRING = "(tttt)";
+    // struct of { int64, int64, int64, int64 }
+    static constexpr const char *TYPE_AS_STRING = "(xxxx)";
 };
 
 template <> struct DBusTypeTrait<Nat64PacketCounters>
 {
-    // struct of { uint64, uint64 }
-    static constexpr const char *TYPE_AS_STRING = "(tt)";
+    // struct of { int64, int64 }
+    static constexpr const char *TYPE_AS_STRING = "(xx)";
 };
 
 template <> struct DBusTypeTrait<Nat64ProtocolCounters>
 {
-    // struct of { struct of { uint64, uint64, uint64, uint64 }
-    //             struct of { uint64, uint64, uint64, uint64 }
-    //             struct of { uint64, uint64, uint64, uint64 }
-    //             struct of { uint64, uint64, uint64, uint64 } }
-    static constexpr const char *TYPE_AS_STRING = "((tttt)(tttt)(tttt)(tttt))";
+    // struct of { struct of { int64, int64, int64, int64 }
+    //             struct of { int64, int64, int64, int64 }
+    //             struct of { int64, int64, int64, int64 }
+    //             struct of { int64, int64, int64, int64 } }
+    static constexpr const char *TYPE_AS_STRING = "((xxxx)(xxxx)(xxxx)(xxxx))";
```

Comment:
Same question here. What is the beneift of using `int64` instead of `uint64`?

Also we should consider if this breaks compatibility? (user of dbus).

Reply:
The idea is following the style guide of https://google.github.io/styleguide/cppguide.html#Integer_Types

Some benefits can be exposing errors (for integer overflow) eariler.

As these API are recently merged, I would expect the impact of the API change is minimal.

Reply:
I read the style guide. It recommends using precise-width integer types over generic `int` and `unsigned it` which we recommend and do it OT as well and I fully agree with it.  👍 

Not sure if there is enough advantage to use `int64` over `uint64` for counters.
- It may lead to confusion, e.g., people may wonder what does it mean and why the counter value can be negative. 
- We can use it to report overflow/undeflow error but not sure how useful and how practical it can be.  Realistically `uint64` counter should not overflow that easily (even if it is increased by `1000` every `usec` it should take more than 500 years).

Reply:
For this PR, I think it is OK to preserve the unsigned types. Given the fact that this values will only be advanced or set to 0. And we won't do other arithmetics on these values.

One idea for using signed values (probably for future, new APIs) is the users is very likely to do arithmetic on these values. Previously, I think it will be more convenient if the users can save this explict type convension, however, seems the compiler won't complain about the convension beetween uint64 and int64.

Based on the style guide, "non negative" should not be implied by the type, and if we are always using signed int unless for some specific cases, I don't think it will confuse people?